File: frontendesiauapp/src/app/composants/services/paiement.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Paiement } from '../models/paiement';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PaiementService {

  
  private baseURL = `${environment.apiUrl}/paiement`;
  private baseURL2 = `${environment.apiUrl}/paiement/etudiant`;
  
  

  constructor( private httpClient: HttpClient) { }

  
  getPaiementList(): Observable<Paiement[]>{
    return this.httpClient.get<Paiement[]>(`${this.baseURL}`);
  }

  createPaiement(paiement: Paiement): Observable<Object>{
    return this.httpClient.post(`${this.baseURL}`, paiement);
  }
  
  updatePaiement(idPaiement: number, paiement: Paiement): Observable<Object>{
    return this.httpClient.put(`${this.baseURL}/${idPaiement}`, paiement);
  }

  deletePaiement(idPaiement: number): Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/${idPaiement}`);
  }
  getPaiementById(idPaiement: number): Observable<Paiement[]>{
    return this.httpClient.get<Paiement[]>(`${this.baseURL}/${idPaiement}`);
  }
  getPaiementListByEtudiant(idEtudiant: number): Observable<Paiement[]>{
    return this.httpClient.get<Paiement[]>(`${this.baseURL2}/${idEtudiant}`);
  }
}

File: backendesiauapp/src/main/java/com/esiau/backendesiauapp/services/PaiementService.java
package com.esiau.backendesiauapp.services;

import java.util.List;
import java.util.Optional;

import com.esiau.backendesiauapp.models.Etudiant;
import com.esiau.backendesiauapp.models.Paiement;



public interface PaiementService {
	
	    public Paiement addPaiement(Paiement paiement);
	    public  List<Paiement> allPaiements();
	    Optional<Paiement> getPaiement(int idPaiement);
	    void deletePaiement(int idPaiement);
	    public Paiement updatePaiement(Paiement paiement);
	    List<Paiement> findPaiementByEtudiant (Etudiant etudiant);

}
